sdk/metric/sdk.go

```diff
@@ -139,6 +143,15 @@ type (
 		labels        *label.Set
 		observed      export.Aggregator
 	}
+
+	// Implementations of MetricsProcessor can be provided as an config option to provide an opportunity
+	// to re-compose metrics labels based on the context when the metrics are recorded.
+	MetricsProcessor interface {
```

Comment:
Does this need to be an interface? It seems like a declared type of `func` would be sufficient here.

Reply:
I think you're right